### PR TITLE
refactor(homeView): reorganize section definitions 

### PR DESCRIPTION
**Pull request by anandaroop**
Part 2 of 2 (see https://github.com/artsy/metaphysics/pull/6009)

As we've been working with the `homeView` code, the split between section definitions and resolvers has felt increasingly artificial to me.

A section **_is_** its metadata and its data resolution — so why do we scatter these pieces across different files?

This PR shows what it would look like to do two things:
- pull each section definition into its own file
- pull each resolver _into_ that new section definition file

Now each section definition completely colocates all of its relevant data and behavior —

https://github.com/artsy/metaphysics/blob/5b979f25066729e8637edb03b607c2c7d2e1b2ec/src/schema/v2/homeView/sections/RecentlyViewedArtworks.ts#L7-L30

We'd end up with something like this:
- _no more_ separate resolver files
- a list of section definition files in `src/schema/v2/homeView/sections`

<pre>
src/schema/v2/homeView
├── HomeViewComponent.ts
├── HomeViewSection.ts
├── __tests__
│   ├── HomeView.test.ts
│   └── HomeViewSection.test.ts
├── helpers
│   ├── getSectionsForUser.ts
│   └── withHomeViewTimeout.ts
├── index.ts
<b>├── sections
│   ├── CuratorsPicksEmerging.ts
│   ├── RecentlyViewedArtworks.ts
│   ├── SimilarToRecentlyViewedArtworks.ts
│   │   (etc)
│   │   (etc)
│   │   (etc)
│   │   (etc)
│   │   (etc)</b>
└── zones
    ├── __tests__
    │   └── legacy.test.ts
    └── legacy.ts
</pre>

We would lose having a single convenient skimmable file of section definitions, that's really the main tradeoff. But feels worth it to me.

These first three commits demonstrate the pattern. If we like  it, I'll continue this for the other sections as well.

### CODE REVIEW

File: src/schema/v2/homeView/sections/CuratorsPicksEmerging.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      viewAll: {
+        href: "/collection/curators-picks-emerging",
+        buttonText: "Browse All Artworks",
+      },
+    },
+    href: "/collection/curators-picks-emerging",
+  },
+  requiresAuthentication: false,
+
+  resolver: withHomeViewTimeout(async (parent, args, context, info) => {
```
Comment (damassi) [inline src/schema/v2/homeView/sections/CuratorsPicksEmerging.ts:48]: +1. This is also idiomatic to the rest of MP! Co-location ftw

**Comment by nickskalkin** [comment]
I like it, +1 from me

**Review by dblandin — DISMISSED** [review]
Awesome! 💯 

Co-locating the data resolution with the rest of a section's metadata makes a lot of sense to me 👍🏼

**Review by MounirDhahri — DISMISSED** [review]
This is a great improvement from what we currently have!

**Comment by anandaroop** [comment]
Great, I'll rebase, finish up the sections migration chore and merge later today

Thanks all 🙏🏽

**Comment by anandaroop** [comment]
Going for the merge before any more rebasing is required.

(Note that prior to merging this PR I noticed some [glitchy behavior](https://artsy.slack.com/archives/C05EQL4R5N0/p1726539832879659) on beta/staging.)

**Pull request merged**